packages/salesforcedx-vscode-core/src/services/sourceStatusSummary.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  Row,
+  Table
+} from '@salesforce/salesforcedx-utils-vscode/out/src/output';
+import { StatusOutputRow } from '@salesforce/source-tracking';
+
+type StatusActualState = 'Deleted' | 'Add' | 'Changed' | 'Unchanged';
+type StatusOrigin = 'Local' | 'Remote';
+
+interface StatusResult {
+  state: string;
+  fullName: string;
+  type: string;
+  filePath?: string;
+  ignored?: boolean;
+  conflict?: boolean;
+  actualState?: StatusActualState;
+  origin: StatusOrigin;
+}
+
+interface FormattedStatusResult {
+  state: string;
+  fullName: string;
+  type: string;
+  filePath?: string;
+  ignored?: string;
+}
+
+export class SourceStatusSummary {
+  constructor(private statusOutputRows: StatusOutputRow[]) {}
+
+  public format(): string {
+    const statusResults = this.statusOutputRows.map(row =>
+      this.resultConverter(row)
+    );
+
+    if (statusResults.length === 0) {
+      return 'No local or remote changes found.';
+    }
+
+    // sort the rows and create a table
+    const sortedStatusResults = statusResults.sort(this.rowSortFunction);
+
+    return new StatusResultsTable(sortedStatusResults).value();
+  }
+
```

Comment:
Next few code blocks [lifted from plugin-source](https://github.com/salesforcecli/plugin-source/blob/main/src/commands/force/source/status.ts#L86), goal is to reuse so that the output stays as close as possible to that from the legacy command (which calls the CLI).